### ISSUE TITLE
Help with maintenance needed?

### ISSUE THREAD

**Issue by czosel** [TomFrost/Jexl#82]
Hi @TomFrost 

First of all (I probably said it before, but it never hurts to say it again): Thanks for your work on this library! Over the past two years, my company's use of Jexl kept growing, so now it's almost critical infrastructure for us :wink: 

This brings me to my actual point: Currently there are some open issues and pull requests, that need merging. Would you appreciate any help with the maintenance? I'd be happy to help out. I already have some experience with the maintenance of OSS projects (see [prettier/plugin-php](https://github.com/prettier/plugin-php/)).

If you'd prefer to keep merge permissions restricted obviously that's totally fine - in that case I'd be happy to hear what your plans for the future of this library are and how I might be able to help out anyway. Thanks! :slightly_smiling_face:

**Comment by czosel** [comment]
Edit: I just realized that I didn't do my research thoroughly (your last activity on the issue tracker is just a few days ago, and you also replied to the issue I raised in #57 over at #60) - I wrote the issue above in the assumption that in the since last october there was no activity, so sorry! Of course, my offer to help still stands.

I'll add a comment to #60 suggesting a concrete solution to the ES6/ES5 issue.

**Comment by TomFrost** [comment]
Hi Christian, thanks for the kind words and generous offer!

I've been hesitant with the PRs because I have a handful of local branches for major functionality changes that all conflict with those. A job change put that work on pause, but I've been picking it back up. Once I get through the critical design changes, I may take you up on the maintainer offer!

**Comment by czosel** [comment]
Hi Tom, thanks for getting back to me about this! I know all too well that OSS work often is the first thing that needs to be put on hold when circumstances change - great to hear that you're picking it back up though! I'm curious about the design changes you have planned :slightly_smiling_face: 
I'll go ahead and close this issue, feel free to reach out when you need anything!

**Comment by TomFrost** [comment]
Thanks!  My major items right now are the option for a binary operator to trigger the evaluation of the left and right sides itself (so, for example, + still evals left and right in parallel, but || and && eval the left first, then choose whether to eval the right), and marrying the Lexer and Parser components together so that the Parser can tell the Lexer which types of token to look for next. That one is necessary to allow the same symbols to be used for unary operators as binary operator. The win there would be that `-` wouldn't require a Lexer hack to determine if it's being used for subtraction or to negate something.

It's also likely this gets an upgrade to Typescript so we don't have parity issues with the DefinitelyTyped repo every time I push a release ;-)

**Comment by czosel** [comment]
:+1: Sounds great!

**Comment by malka-ch** [comment]
any updates?
